### PR TITLE
[CDAP-2908] add loading icon to the make request button in service program

### PR DESCRIPTION
**Pull request by elfenheart**
service

screen shot 2015-11-04 at 11 39 34 am

screen shot 2015-11-04 at 11 40 00 am

### CODE REVIEW

File: cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js [hunk 1]
```diff
@@ -108,8 +110,12 @@ angular.module(PKG.name + '.feature.services')
       }
 
       dataSrc.request(requestObj)
-        .then(function(res) {
+        .then(function success(res) {
           vm.response = res;
+          vm.loading = false;
+        }, function error(err) {
+          vm.response = 'Some error has occured: ' + err;
```
Comment (ajainarayanan) [inline cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js]: This should be ```err.data``` right?

File: cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js [hunk 2]
```diff
@@ -108,8 +110,12 @@ angular.module(PKG.name + '.feature.services')
       }
 
       dataSrc.request(requestObj)
-        .then(function(res) {
+        .then(function success(res) {
           vm.response = res;
+          vm.loading = false;
```
Comment (ajainarayanan) [inline cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js]: So we should be able to show some standard response when the POST is successful. Right now backend just sends 200 and we don't know if something actually happened and is successful or now.

File: cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js (hunk 1 above)
Comment (elfenheart) [inline cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js]: My thought was to just show the entire error object for ease of debugging. What do you think? @ajainarayanan

Comment (ajainarayanan) [inline cdap-ui/app/features/services/controllers/tabs/runs/tabs/status/make-request-ctrl.js]: @elfenheart Its just showing [Object object] in the browser. I think the backend is returning only string as error (be it stack trace too). We could probably stick to just string.

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Pull request merged**